src/test_application/flight.cpp

```diff
@@ -127,20 +129,30 @@ void testapp::test_flight(std::unique_ptr<OSPMagnum>& pMagnumApp,
 
     // Create the camera entity
     ActiveEnt camera = rScene.hier_create_child(rScene.hier_get_root(),
-                                                       "Camera");
+        "Camera");
     auto &cameraTransform = rScene.reg_emplace<ACompTransform>(camera);
     auto &cameraComp = rScene.get_registry().emplace<ACompCamera>(camera);
 
     cameraTransform.m_transform = Matrix4::translation(Vector3(0, 0, 25));
     rScene.reg_emplace<ACompFloatingOrigin>(camera);
 
-    cameraComp.m_viewport
-            = Vector2(Magnum::GL::defaultFramebuffer.viewport().size());
-    cameraComp.m_far = 1u << 24;
-    cameraComp.m_near = 1.0f;
-    cameraComp.m_fov = 45.0_degf;
+    {
+        auto& resources = rScene.get_context_resources();
+        Magnum::Vector2i viewSize = Magnum::GL::defaultFramebuffer.viewport().size();
+
+        // Fetch primary render target (an offscreen FBO)
+        osp::DependRes<Magnum::GL::Framebuffer> fbo =
+            resources.get<Magnum::GL::Framebuffer>("offscreen_fbo");
```

Comment:
How else should we configure things like this? Wrap them in an additional abstraction layer?

Reply:
The thing is, the whole purpose of Magnum is to provide that abstraction layer, so I'm not sure what we'd gain by building more abstraction on top. Ultimately a camera added to the scene is going to need to specify what surface it should render to.

I don't really have a good sense for what sort of abstraction will allow us to keep the flexibility while supporting the future architecture, since we don't yet have multiple scenes running at the same time with different rendering requirements.

Reply:
> The thing is, the whole purpose of Magnum is to provide that abstraction layer, so I'm not sure what we'd gain by building more abstraction on top. 

If systems run on the worker thread (which they would in the future), then there's no way they can create GL objects. 

> How else should we configure things like this? This is configuration that'll vary from scene to scene. Wrap them in an additional abstraction layer?

Take a look at how Newton is integrated: all other systems just create generic components, and SysNewton creates (and synchronizes) the Newton bodies with those generic properties.

> Ultimately a camera added to the scene is going to need to specify what surface it should render to.

Is there a good reason for this? This means we're iterating cameras to render? What would happen if more than one camera render to the same target? 

I've never seen any other application have the cameras themselves that specify where they render to. Blender has the scene specify a single active camera. Urho3D has viewports specify a camera, which draws onto a render target.

Reply:
Yeah a viewport abstraction or something similar may provide the abstraction to do what you're describing